acp/ACP.md

```diff
@@ -0,0 +1,419 @@
+The way Zanzibar works is that you write a tuple that includes subject, relation, and object. Then, you check if a permission exists and run an expand call that acts as a reverse lookup over a user set. It operates as a Consistent and Partition-Tolerant System.
+
+### Zanzi
+However the the Zanzibar API is centralized, so we (Source Network) created a decentralized implementation of Zanzibar called **Zanzi**. Which is powered by our SourceHub trust protocol.
+
+### SourceHub Module
+DefraDB wraps the **local** and **remote** SourceHub ACP Modules to bring all that magic to DefraDB.
+
+Inorder to setup the relation based access control, SourceHub requires an agreed upon contract which models the **relations**, **permissions**, and **actors**. That contract is refered to as a **SourceHub Policy**. The policy model's all the **relations** and **permissions** under a **resource**.
+A **resource** corresponds to that "thing" that we want to gate the access control around. This can be a `Type`, `Container`, `Schema`, `Shape` or anything that has Objects that need access control. Once the policy is finalized, it has to be uploaded to the **SourceHub Module** so it can be used.
+Once the **Policy** is uploaded to the **SourceHub Module** then an **Actor** can begin registering the **Object** for access control by linking to a **Resource** that exists on the uploaded **Policy**.
+After the **Object** is registered successfully, the **Actor** will then get a special built-in relation with that **Object** called the **"owner"** relation. This relation is given to the **Registerer** of an **Object**.
+Then an **Actor** can issue **Check** calls to see if they have access to an **Object**.
+
+## Document Access Control (DAC)
+In DefraDB's case we wanted to gate access control around the **Documents** that belonged to a specific **Collection**. Here, the **Collection** (i.e. the type/shape of the **Object**) can be thought of as the **Resource**, and the **Documents** are the **Objects**.
+
+
+## Field Access Control (FAC) (coming soon)
+We also want the ability to do a more granular access control than just DAC. Therefore we have **Field** level access control for situations where some fields of a **Document** need to be private, while others do not. In this case the **Document** becomes the **Resource** and the **Fields** are the **Objects** being gated.
+
+
+## Admin Access Control (AAC) (coming soon)
+We also want to model access control around the **Admin Level Operations** that exist in **DefraDB**. In this case the entire **Database** would be the **Resource** and the **Admin Level Operations** are the **Objects** being gated.
+
+A non-exhastive list of some operations only admins should have access for:
+- Ability to turnoff ACP
+- Ability to interact with the P2P system
+
+## SourceHub Policies Are Too Flexible
```

Comment:
Haha I would argue that this paragraph is perhaps the most useful, this is perhaps the whole reason for me writing this entire document, because people requested to know the "why" behind needing DPI compliance. This paragraph answers that. Happy to suggestions on how we want to improve this section.